apps/cms/scripts/directus.mjs

```diff
@@ -1,12 +1,70 @@
 import { Directus } from '@directus/sdk'
 
-export async function configureDirectus() {
-  const client = new Directus(process.env.PUBLIC_URL)
+export async function setupSDK(config) {
+  const directus = new Directus(process.env.PUBLIC_URL);
 
-  await client.auth.login({
-    email: process.env.ADMIN_EMAIL,
-    password: process.env.ADMIN_PASSWORD,
-  })
+  // But, we need to authenticate if data is private
+  let authenticated = false;
 
-  return client
+  // Try to authenticate with token if exists
+  await directus.auth
+    .refresh()
+    .then(() => {
+      authenticated = true;
+    })
+    .catch(() => { });
+
+  // Let's login in case we don't have token or it is invalid / expired
+  while (!authenticated) {
+    await directus.auth
+      .login(config)
+      .then(() => {
+        authenticated = true;
+      })
+      .catch(() => {
+        window.alert('Invalid credentials');
+      });
+  }
+
+  return directus
+}
+
+export async function setAccessToken(directus) {
+  const me = await directus.users.me.read();
+
+  await directus.users.updateOne(me.id, {
+    token: process.env.ADMIN_ACCESS_TOKEN,
+  });
+}
+
+export async function setFilePermission(directus) {
```

Comment:
_Technically_, we don't need the file permissions to be public if you use the GCP bucket and its read-access is public. But could be good to keep it around?